dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldPath.java

```diff
@@ -98,6 +98,15 @@ public FieldPath( String name, List<String> path )
         this.transformers = new ArrayList<>();
     }
 
+    public FieldPath( String name, List<String> path, boolean exclude, boolean preset )
+    {
+        this.name = name;
+        this.path = path;
+        this.exclude = exclude;
+        this.preset = preset;
+        this.transformers = new ArrayList<>();
```

Comment:
I just see `this.transformers` from the constructor above in the diff - I think it is best if we can limit ourselves to one constructor that actually sets the fields and the others just calling that one. If that isn't possible usually it is a good indication for some bad practice going on.

Reply:
Yep, issue is that the properties are final and so calling another constructor is not allowed.

Reply:
Hm... any constructor must set final fields and it is possible to call constructors to do so. So this might be a case where we just got into having multiple constructors that set fields because there is no constructor that does this for all fields which we could call. Anyhow, just wanted to point this out in case the code could make use of calling another constructor instead.